app/components/mandatarissen/update-state.js

```diff
@@ -7,23 +7,39 @@ import { showErrorToast, showSuccessToast } from 'frontend-lmb/utils/toasts';
 
 export default class MandatarissenUpdateState extends Component {
   @tracked newStatus = null;
-  @tracked date = new Date();
+  @tracked date = null;
+  @tracked selectedBeleidsdomeinen = [];
+  @tracked selectedFractie = null;
+  @tracked bestuursorganenForFractie = [];
+  @tracked rangorde = null;
 
   @service mandatarisStatus;
   @service store;
   @service toaster;
 
   constructor() {
     super(...arguments);
-    this.newStatus = this.mandatarisStatus.endedState;
+    this.load.perform();
+  }
+
+  get loading() {
+    return this.load.isRunning;
+  }
+
+  @dropTask
+  *load() {
+    this.newStatus = this.args.mandataris.status;
+    this.date = new Date();
+    this.selectedBeleidsdomeinen = this.args.mandataris.beleidsdomein.slice();
+    this.rangorde = this.args.mandataris.rangorde;
+    this.selectedFractie = yield (yield this.args.mandataris.heeftLidmaatschap)
```

Comment:
painful, isn't this solvable as well with the get('..') instead? But then again, wouldn't really make it much better.

Reply:
indeed